gridded_etl_tools/utils/convenience.py

```diff
@@ -632,4 +635,31 @@ def span_to_timedelta(cls) -> np.timedelta64:
             cls.SPAN_MONTHLY: np.timedelta64(1, "M"),
             cls.SPAN_YEARLY: np.timedelta64(1, "Y"),
         }
-        return span_to_td[cls.temporal_resolution()]
+        return span_to_td
+
+    def get_random_coords(self, dataset: xr.Dataset) -> dict[str, Any]:
+        """
+        Derive a dictionary of random coordinates, one for each coordinate in the input dataset
+
+        Parameters
+        ----------
+        dataset
+            An Xarray dataset
+        """
+        coords_dict = {}
+        # We select from dims, not coords because inserts drop all non-time_dim coords
+        for dim in dataset.dims:
+            coords_dict.update({dim : random.choice(dataset[dim].values)})
+        return coords_dict
+    
+    @property
+    def extreme_values_by_unit(self):
```

Comment:
If these are intended to be the same for all datasets (basically a global var) they probably belong defined at the module level, then accessed by `convenience.extreme_values_by_unit` rather than `convenience.Convenience.extreme_values_by_unit`/`self.extreme_values_by_unit`

Reply:
I’m hesitant to introduce a new pattern like this into the `tools` repo. The convenience of calling every property with `self` is a major plus of the current setup, even when those properties are in effect static.

Reply:
It's just a bit different than the other properties because it's not meant to be override-able per dataset, but I guess it's fine if it lives here